### PR TITLE
feat: introduce WithQPS and WithBurst functions

### PR DESCRIPTION
**Pull request by thapabishwa**
@chen-keinan 
Unfortunately I couldn't create an issue on this repo like we discussed. 

As such, I have created this PR to reflect the changes that I made on my end to while I encountered the issue.
I believe that might help you to pin point the issue.

### CODE REVIEW

File: pkg/k8s/k8s.go [hunk 1]
```diff
@@ -165,6 +180,35 @@ func getCluster(clientConfig clientcmd.ClientConfig, restMapper meta.RESTMapper,
...
+		Setting burst to 2000
+		Setting QPS to 2000
+		kubeconfig QPS: 2000 (manually set values)
+		kubeconfig Burst: 2000 (manually set values)
+		Current namespace: default
+		Scanning cluster
+		Time taken: 89.546083ms
+		Name: coredns, Kind: Deployment, Namespace: kube-system, Images: [registry.k8s.io/coredns/coredns:v1.10.1]
+	*/
+
```
Comment (chen-keinan) [inline pkg/k8s/k8s.go:211]: thanks for example , I think this comment can be removed

File: pkg/k8s/k8s.go [hunk 2]
```diff
@@ -139,6 +139,21 @@ func WithKubeConfig(kubeConfig string) ClusterOption {
 	}
 }
 
+func WithQPS(qps float32) ClusterOption {
+	return func(o *genericclioptions.ConfigFlags) {
+		fmt.Println("Setting QPS to", qps)
```
Comment (chen-keinan) [inline pkg/k8s/k8s.go:144]: I think the `fmt.Println`  can be removed

File: pkg/k8s/k8s.go [hunk 3]
```diff
@@ -139,6 +139,21 @@ func WithKubeConfig(kubeConfig string) ClusterOption {
...
+		o.WithDiscoveryQPS(qps)
+	}
+}
+
+// WithBurst sets the burst limit for the client
+func WithBurst(burst int) ClusterOption {
+	return func(o *genericclioptions.ConfigFlags) {
+		fmt.Println("Setting burst to", burst)
```
Comment (chen-keinan) [inline pkg/k8s/k8s.go:152]: I think the `fmt.Println`  can be removed

**Review by chen-keinan — COMMENTED** [review]
@thapabishwa thank you for pointing it out. change lgtm 🚀  I have added couple of comments

**Comment by thapabishwa** [comment]
Closed in favor of #263.

Changes in that PR can successfully update QPS/Burst values of the clients.

**Pull request closed without merging**